src/module/actor/pokemon/document.js

```diff
@@ -354,27 +354,51 @@ class PTUPokemonActor extends PTUActor {
 
     // TODO: Implement rules for capability changing items
     _calcCapabilities() {
-        const capabilities = duplicate(this.species?.system?.capabilities ?? {});
-        if (!capabilities) return {};
+        const movementCapabilities = ["overland", "swim", "burrow", "sky", "levitate", "teleporter"]
+        const numericNonMovementCapabilities = ["highJump", "longJump", "power", "weightClass"]
+        const stringArrayCapabilities = ["naturewalk", "other"]
+
+        const speciesCapabilities = duplicate(this.species?.system?.capabilities ?? {});
+        const finalCapabilities = {}
+        if (!speciesCapabilities) return {};
 
         const speedCombatStages = this.system.stats.spd.stage.value + this.system.stats.spd.stage.mod;
         const spdCsChanges = speedCombatStages > 0 ? Math.floor(speedCombatStages / 2) : speedCombatStages < 0 ? Math.ceil(speedCombatStages / 2) : 0;
-        const capabilityMod = Number(this.system.modifiers.capabilities.all ?? 0);
-        for (const key of Object.keys(capabilities)) {
-            if (key == "highJump" || key == "longJump" || key == "power" || key == "weightClass" || key == "naturewalk" || key == "other") continue;
-            if (capabilities[key] > 0) {
-                capabilities[key] = Math.max(capabilities[key] + spdCsChanges + capabilityMod, capabilities[key] > 1 ? 2 : 1)
-                if (this.rollOptions.conditions?.["slowed"]) capabilities[key] = Math.max(1, Math.floor(capabilities[key] * 0.5));
-                capabilities[key] = Math.max(1, capabilities[key] + (this.system.modifiers.capabilities?.[key] ?? 0));
+        const omniMovementMod = Number(this.system.modifiers.capabilities.all ?? 0);
```

Comment:
Instead of `Number(this.system.modifiers.capabilities.all ?? 0)`
it is better to use `Number(this.system.modifiers.capabilities.all) || 0`

The initial statement: If capabilities.all is null, become a 0.
Revised: if capabilites.all is NaN, become a 0.

Reply:
Thanks for the advice.